fuse_models/test/test_unicycle_2d_state_cost_function.cpp

```diff
@@ -56,85 +56,105 @@ TEST(CostFunction, evaluateCostFunction)
-  const double vel_linear1[] = {1.0, 0.0};
-  const double vel_yaw1[] = {1.570796327};
-  const double acc_linear1[] = {1.0, 0.0};
-
-  const double position2[] = {0.105, 0.0};
-  const double yaw2[] = {0.1570796327};
-  const double vel_linear2[] = {1.1, 0.0};
-  const double vel_yaw2[] = {1.570796327};
-  const double acc_linear2[] = {1.0, 0.0};
-
-  const double* parameters[] =
-  {
-    position1, yaw1, vel_linear1, vel_yaw1, acc_linear1,
-    position2, yaw2, vel_linear2, vel_yaw2, acc_linear2
-  };
-
-  fuse_core::Vector8d residuals;
-
-  const auto& block_sizes = cost_function.parameter_block_sizes();
-  const auto num_parameter_blocks = block_sizes.size();
-
-  const auto num_residuals = cost_function.num_residuals();
-
-  std::vector<fuse_core::MatrixXd> J(num_parameter_blocks);
-  std::vector<double*> jacobians(num_parameter_blocks);
-
-  for (size_t i = 0; i < num_parameter_blocks; ++i)
-  {
-    J[i].resize(num_residuals, block_sizes[i]);
-    jacobians[i] = J[i].data();
-  }
-
-  EXPECT_TRUE(cost_function.Evaluate(parameters, residuals.data(), jacobians.data()));
-
-  // We cannot use std::numeric_limits<double>::epsilon() tolerance because with the expected state2 above the residuals
-  // are not zero for position2.x = -4.389e-16 and yaw2 = -2.776e-16
-  EXPECT_MATRIX_NEAR(fuse_core::Vector8d::Zero(), residuals, 1e-15);
-
-  // Check jacobians are correct using a gradient checker
-  ceres::NumericDiffOptions numeric_diff_options;
-  ceres::GradientChecker gradient_checker(&cost_function, nullptr, numeric_diff_options);
-
-  // We cannot use std::numeric_limits<double>::epsilon() tolerance because the worst relative error is 5.26356e-10
-  ceres::GradientChecker::ProbeResults probe_results;
-  // TODO(efernandez) probe_results segfaults when it's destroyed at the end of this TEST function, but Probe actually
-  // returns true and the jacobians are correct according to the gradient checker numeric differentiation
-  // EXPECT_TRUE(gradient_checker.Probe(parameters, 1e-9, &probe_results)) << probe_results.error_log;
-
-  // Create cost function using automatic differentiation on the cost functor
-  ceres::AutoDiffCostFunction<fuse_models::Unicycle2DStateCostFunctor, 8, 2, 1, 2, 1, 2, 2, 1, 2, 1, 2>
-      cost_function_autodiff(new fuse_models::Unicycle2DStateCostFunctor(dt, sqrt_information));
-
-  // Evaluate cost function that uses automatic differentiation
-  std::vector<fuse_core::MatrixXd> J_autodiff(num_parameter_blocks);
-  std::vector<double*> jacobians_autodiff(num_parameter_blocks);
-
-  for (size_t i = 0; i < num_parameter_blocks; ++i)
-  {
-    J_autodiff[i].resize(num_residuals, block_sizes[i]);
-    jacobians_autodiff[i] = J_autodiff[i].data();
-  }
-
-  EXPECT_TRUE(cost_function_autodiff.Evaluate(parameters, residuals.data(), jacobians_autodiff.data()));
-
-  const Eigen::IOFormat HeavyFmt(
-      Eigen::FullPrecision, 0, ", ", ";\n", "[", "]", "[", "]");
+  std::random_device dev;
+  std::mt19937 gen(dev());
```

Comment:
If the test fails, would we be able to reproduce it?

I think it'd be better if we recode the seed. I've never than that, but I believe we could print the seed:
```suggestion
  const auto seed = dev();
  std::mt19937 gen(seed);
```

and then pass `<< seed` to all the checks below that depend on the seed, so we can see what seed was used and then we can easily reproduce the test.

Another option is to just use a fixed seed.